.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,18 @@
+name: Docker Image CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
```

Comment:
@eternaltyro  Do we need to build a docker image on every PR? I think doing it only when this gets to master will be better